src/main/java/org/checkerframework/specimin/UnsolvedSymbolVisitor.java

```diff
@@ -850,6 +907,44 @@ public String qualifiedNameToFilePath(String qualifiedName) {
     return filePath;
   }
 
+  /**
+   * Given a method call, this method checks if that method call is declared in the currently
+   * visiting class.
+   *
+   * @param method the method call to be checked
+   * @return true if method is declared in the current clases
+   */
+  public boolean declaredInCurrentClass(MethodCallExpr method) {
+    for (Set<MethodDeclaration> methodDeclarationSet : declaredMethod) {
+      for (MethodDeclaration methodDeclared : methodDeclarationSet) {
+        if (!methodDeclared.getName().asString().equals(method.getName().asString())) {
+          continue;
+        }
+        List<String> methodTypesOfParameters = getArgumentsFromMethodCall(method);
+        NodeList<Parameter> methodDeclaredParameters = methodDeclared.getParameters();
+        List<String> methodDeclaredTypesOfParameters = new ArrayList<>();
+        for (Parameter parameter : methodDeclaredParameters) {
+          try {
+            ResolvedType parameterTypeResolved = parameter.getType().resolve();
+            if (parameterTypeResolved.isPrimitive()) {
+              methodDeclaredTypesOfParameters.add(parameterTypeResolved.asPrimitive().name());
+            } else if (parameterTypeResolved.isReferenceType()) {
+              methodDeclaredTypesOfParameters.add(
+                  parameterTypeResolved.asReferenceType().getQualifiedName());
+            }
+          } catch (UnsolvedSymbolException e) {
+            // UnsolvedSymbolVisitor will not create any synthetic class at this iteration.
+            return false;
+          }
+        }
+        if (methodDeclaredTypesOfParameters.equals(methodTypesOfParameters)) {
+          return true;
```

Comment:
@LoiNguyenCS Sorry to bring this up now, when this has already been merged, but I don't think that this logic is correct. In particular, it fails to take into account subtyping: it is legal to call a method with arguments that are subtypes of the declared types of the method, but this method will return false in that case. For example, consider:
```
class A {}
class B extends A {}

class C {
   void test(A a) { ... }
...
   test(new B());
}
```
The call to `test(new B())` is legal and typechecks, but this method would miss that this call is to a method defined in the class. The problem is that the argument and parameter lists don't exactly match up - each argument is a subtype.

I'm unsure if this is causing any of our problems (I observed it while debugging #152, but it is not the proximate cause of that issue I don't think), but we need to come up with a strategy to handle this case properly.